auth/consumer.go

```diff
@@ -0,0 +1,20 @@
+package auth
+
+// Consumer represents an API consumer.
+type Consumer struct {
+	ID     int64   `json:"id"`
+	Grants []Grant `json:"grants"`
+}
+
+// HasGrants checks if a consumer possess any of a given set of grants
+func (c *Consumer) HasGrants(grants ...Grant) bool {
```

Comment:
As discussed with @nicklanng, perhaps use the word `Any` in this method name to make it clear that we're happy with _any_ of the grants to be held, to prevent confusion at the early return.